scene/2d/animated_sprite_2d.cpp

```diff
@@ -228,11 +228,10 @@ void AnimatedSprite2D::_notification(int p_what) {
 
 			RID ci = get_canvas_item();
 
-			Size2i s;
-			s = texture->get_size();
+			Size2i s = texture->get_size();
```

Comment:
Texture's get_size() returns Size2().
So, wouldn't it be better to change Size2i to Size2 instead,so that you don't need to cast it to Size2() below?

Reply:
> Texture's get_size() returns Size2().

Yeah, I know. However, AFAIK currently every `Texture`-derived type returns just `Size2(get_width(), get_height())` so it consists of ints (`get_size()` is virtual so it could potentially be overridden though). In many places in the code base there's something like `Size2i s = some_texture.get_size();` so I'd say `get_size()` is kinda expected to really return `Size2i`. Thus I just kinda wanted to be consistent with the code base and make the intention clear by casting to `Size2` explicitly.
I can of course change it if what you've suggested is preferred.

Reply:
You are correct too.
Additionally, in the same Animated Sprite2d Class, there is the similar code,using Size2.

https://github.com/godotengine/godot/blob/9013771b53cdbc6451f2645f3618ad59d90a2fb1/scene/2d/animated_sprite_2d.cpp#L93-L104

Someone,from core should confirm it though,what's the more correct thing to do. :)

Reply:
I've made these changes.